File: 34-FindFirstAndLastPositionOfElementInSortedArray/34-FindFirstAndLastPositionOfElementInSortedArray.java
// Last updated: 7/12/2026, 6:25:11 PM
class Solution {
    public int[] searchRange(int[] nums, int target) {
        
        int start = binarySearch(nums,target,true);
        int end = binarySearch(nums,target,false);
        
        int[] ans = new int[2];
        
        ans[0] = start;
        ans[1] = end;
        
        return ans;
        
        
        
    }
    
    int binarySearch(int[] nums, int target, boolean searchFirstOccurence){
        
        int start = 0;
        int end = nums.length - 1;
        int ans = -1;
        
        while(start <= end){
            int mid = start + (end - start) / 2;
            
            if(nums[mid] < target) start = mid + 1;
            else if(nums[mid] > target) end = mid - 1;
            else if(nums[mid] == target) {
                ans = mid;
                
                if(searchFirstOccurence) end = mid - 1;
                else start = mid + 1;
            }
            
            
        }
        
        return ans;
        
    }
}

File: 34-FindFirstAndLastPositionOfElementInSortedArray/34-FindFirstAndLastPositionOfElementInSortedArray.py
# Last updated: 7/12/2026, 6:25:07 PM
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        s = self.binarySearch(nums,target, True)
        e = self.binarySearch(nums,target, False)

        return [s, e]

    def binarySearch(self, nums: List[int], target: int, firstOccurence: bool) -> int:
        l, h = 0, len(nums)-1
        ans = -1

        while l <= h:
            mid = (l + h) // 2
            if nums[mid] > target: h = mid - 1
            elif nums[mid] < target: l = mid + 1
            else:
                ans = mid

                if firstOccurence: h = mid - 1
                else: l = mid + 1

                

        
        return ans


        